.pre-commit-config.yaml

```
@@ -34,6 +34,7 @@ repos:
           - pip==20.3.4
           - build==1.0.0
           - pyproject_hooks==1.0.0
+          - pytest>=7.2.0
```

Generally requirements should be pinned to ensure stability. With several issues fixed in 7.4.2, I think we should use that as a mark.

```suggestion
          - pytest==7.4.2
```

---

I agree, since I let RenovateBot or similar tools keep the dependencies up-to-date in my own projects.

But I simply copied the constraint from the `project.optional-dependencies` section in `pyproject.toml`. If pytest is pinned in the pre-commit configuration, then it should be pinned to the same version in `pyproject.toml`.

---

So far, none of the dependencies in `pyproject.toml` are pinned to a fixed version. Of course, I could pin them, but that's a change in how your project handles dependencies that's unrelated to my bugfix. Do you want this change mixed into this PR?

---

I guess it wouldn't hurt if you just pinned it in the pre-commit configuration right now. There is a proposal to update to newer dependencies in #1927, but it was held back by a refactoring idea. We'll sort it out in that context.